src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java

```diff
@@ -128,36 +125,93 @@ private static boolean isSupportedVideo(String mimeType) {
+			return false;
+		}
+
+		if (formatInfo.duration() == null) {
+			return false;
+		}
+
+		var videoInfo = mediaInfo.video();
+		if (videoInfo == null) {
+			return false;
+		}
+
+		return isSizeCompliant(videoInfo.width(), videoInfo.height())
+				&& videoInfo.frameRate() <= MAX_VIDEO_FRAMES
+				&& VP9_CODEC.equals(videoInfo.codec())
+				&& Float.parseFloat(formatInfo.duration()) <= MAX_VIDEO_DURATION_SECONDS
+				&& mediaInfo.audio() == null
+				&& formatInfo.format().startsWith(MATROSKA_FORMAT)
+				&& Long.parseLong(formatInfo.size()) <= MAX_VIDEO_FILE_SIZE;
 	}
 
 	/**
 	 * Convenience method to retrieve multimedia information of a file.
 	 *
 	 * @param file the video to check
 	 * @return passed-in video's multimedia information
-	 * @throws CorruptedVideoException if an error occurred retrieving video information
+	 * @throws CorruptedFileException if an error occurred retrieving file information
+	 * @throws InterruptedException if the current thread is interrupted while retrieving file info
 	 */
-	private static MultimediaInfo retrieveMultimediaInfo(File file) throws CorruptedVideoException {
+	static MultimediaInfo retrieveMultimediaInfo(File file) throws CorruptedFileException, InterruptedException {
+		var command = new String[] {
+				"ffprobe",
+				"-hide_banner",
+				"-v", "quiet",
+				"-print_format", "json",
+				"-show_format",
+				"-show_streams",
+				file.getAbsolutePath()
+		};
+
 		try {
-			return new MultimediaObject(file, PathLocator.INSTANCE).getInfo();
-		} catch (EncoderException e) {
-			throw new CorruptedVideoException("The video could not be processed successfully", e);
+			var lines = ProcessHelper.executeCommand(command);
+			var body = String.join("\n", lines);
+
+			return GSON.fromJson(body, MultimediaInfo.class);
+		} catch (ProcessException | JsonSyntaxException e) {
+			throw new CorruptedFileException("The file could not be processed successfully", e);
+		}
+	}
+
+	record MultimediaInfo(List<StreamInfo> streams, @Nullable FormatInfo format) {
+		@Nullable StreamInfo audio() {
+			return streams.stream()
+					.filter(s -> s.type == CodecType.AUDIO)
+					.findFirst()
+					.orElse(null);
+		}
+
+		@Nullable StreamInfo video() {
+			return streams.stream()
+					.filter(s -> s.type == CodecType.VIDEO)
+					.findFirst()
+					.orElse(null);
 		}
 	}
+	record StreamInfo(@SerializedName("codec_name") String codec, @SerializedName("codec_type") CodecType type, int width, int height, @SerializedName("avg_frame_rate") String frameRateRatio) {
+		float frameRate() {
+			if (frameRateRatio.contains("/")) {
+				var ratio = frameRateRatio.split("/");
+				return Float.parseFloat(ratio[0]) / Float.parseFloat(ratio[1]);
+			} else {
+				return Float.parseFloat(frameRateRatio);
+			}
+		}
+	}
+	private enum CodecType {
+		@SerializedName("video") VIDEO,
+		@SerializedName("audio") AUDIO
+	}
+	record FormatInfo(@SerializedName("format_name") String format, @Nullable String duration, String size) {}
```

Comment:
Can we deserialize them to the type we need or is it troublesome?
```suggestion
	record FormatInfo(@SerializedName("format_name") String format, @Nullable float duration, long size) {}
```